app/models/user.rb

```diff
@@ -505,6 +505,18 @@ def latest_location
     recent_locations.last
   end
 
+  def self.recently_active_users(limit = 5, order = 'last_updated DESC')
+    Rails.cache.fetch('users/active', expires_in: 24.hours) do
+      User.select('rusers.username, rusers.status, rusers.id, MAX(node_revisions.timestamp) AS last_updated')
+        .joins("INNER JOIN `node_revisions` ON `node_revisions`.`uid` = `rusers`.`id` ")
+        .where("node_revisions.status = 1")
+        .where("rusers.status = 1")
+        .group('rusers.id')
+        .order(order)
+        .limit(limit)
```

Comment:
Here, i wonder -- is 5 enough? What if we changed the limit to 15? Or, alternatively, what if there were a way to narrow this list based on something specific to the current user? But, that might increase the complexity of the query and therefore slow it down a lot. 

Finally, i wonder if we could add caching to this query. That could make this really speedy and it seems OK for there to be at least an hour long cache here, maybe? Look at this example - 

https://github.com/publiclab/plots2/blob/cfba9ef309198ca0731f9c0d54214816c8915278/app/controllers/stats_controller.rb#L5

This can be added in followup, so I'll go ahead and merge this now, but these are maybe some more refinements we can make -- and they're simple enough that maybe they could be an FTO too!

Reply:
> What if we changed the limit to 15?

@jywarren Setting the limit to 15 sounds like a good idea. 


> Finally, i wonder if we could add caching to this query. That could make this really speedy and it seems OK for there to be at least an hour long cache here, maybe? Look at this example -

The query already has a cache or is there somewhere else I'm supposed to add it? The cache has an expiration time of 24 hours so I'll change it to an hour.

https://github.com/publiclab/plots2/blob/c76e4859dcb69090ccf31ed8fe6140e4eaa567d2/app/models/user.rb#L509